File: auth-data-server/src/main/java/com/cartobucket/auth/rpc/server/entities/ScopeReference.java
package com.cartobucket.auth.rpc.server.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;

import java.util.UUID;

@Entity
@Table(uniqueConstraints = { @UniqueConstraint(columnNames = { "scopeId", "resourceId" }) })
public class ScopeReference {
    public enum ScopeReferenceType {
        AUTHORIZATION_SERVER,
        APPLICATION,
        APPLICATION_SECRET,
        CLIENT
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public UUID getScopeId() {
        return scopeId;
    }

    public void setScopeId(UUID scopeId) {
        this.scopeId = scopeId;
    }

    public UUID getResourceId() {
        return resourceId;
    }

    public void setResourceId(UUID resourceId) {
        this.resourceId = resourceId;
    }

    public ScopeReferenceType getScopeReferenceType() {
        return scopeReferenceType;
    }

    public void setScopeReferenceType(ScopeReferenceType scopeReferenceType) {
        this.scopeReferenceType = scopeReferenceType;
    }

    @Id
    @GeneratedValue
    private UUID id;

    private UUID scopeId;

    private UUID resourceId;
    private ScopeReferenceType scopeReferenceType;
}

File: auth-data-postgres-client/src/main/kotlin/com/cartobucket/auth/postgres/client/entities/ScopeReference.kt
package com.cartobucket.auth.postgres.client.entities

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.util.UUID

@Entity
@Table(
//        uniqueConstraints = [UniqueConstraint(columnNames = ["scopeId", "resourceId"])]
)
class ScopeReference {
    enum class ScopeReferenceType {
        AUTHORIZATION_SERVER,
        APPLICATION,
        APPLICATION_SECRET,
        CLIENT,
    }

    @Id
    @GeneratedValue
    var id: UUID? = null

    var scopeId: UUID? = null

    var resourceId: UUID? = null

    var scopeReferenceType: ScopeReferenceType? = null
}
